AfsExceptionExclusiveAttemptConflict: File is already used by a different exclusive user on restart

```
Caused by: one.microstream.afs.exceptions.AfsExceptionExclusiveAttemptConflict: File is already used by a different exclusive user: one.microstream.afs.types.AFile$Default@6327b54d("storage/channel_3/transactions_3.sft"). Exclusive user: (java.lang.Thread@8674c87) Thread[MicroStream-StorageChannel-3,5,]. Attempting user: (java.lang.Thread@6a0b3fb) Thread[MicroStream-StorageChannel-3,5,main].
	at one.microstream.afs.types.AccessManager$Default$2.handleExclusiveAttemptConflict(AccessManager.java:776)
	at one.microstream.afs.types.AccessManager$Default.internalUseWriting(AccessManager.java:664)
	at one.microstream.afs.types.AccessManager$Default.useWriting(AccessManager.java:497)
	at one.microstream.afs.types.AFile.useWriting(AFile.java:108)
	at one.microstream.afs.types.AFS.applyWriting(AFS.java:272)
	at one.microstream.afs.types.AFS.applyWriting(AFS.java:262)
	at one.microstream.afs.types.AFile.ensureExists(AFile.java:157)
	at one.microstream.storage.types.StorageFileManager$Default.createTransactionsFile(StorageFileManager.java:1117)
	at one.microstream.storage.types.StorageFileManager$Default.readTransactionsFile(StorageFileManager.java:727)
	at one.microstream.storage.types.StorageFileManager$Default.readStorage(StorageFileManager.java:712)
	at one.microstream.storage.types.StorageChannel$Default.readStorage(StorageChannel.java:745)
	at one.microstream.storage.types.StorageChannelTaskInitialize$Default.internalProcessBy(StorageChannelTaskInitialize.java:195)
	at one.microstream.storage.types.StorageChannelTaskInitialize$Default.internalProcessBy(StorageChannelTaskInitialize.java:1)
	at one.microstream.storage.types.StorageChannelTask$Abstract.processBy(StorageChannelTask.java:222)
	at one.microstream.storage.types.StorageChannel$Default.work(StorageChannel.java:422)
	at one.microstream.storage.types.StorageChannel$Default.run(StorageChannel.java:509)
	at java.lang.Thread.run(Thread.java:748)
```

After fixing #279 that exception gets reproducible thrown after restarting a storage several times if the continuous Backup is enabled.

The Exception is thrown because the AccessManager instance does not deregister the affected file on shutdown. At the next restart the error is caused.

--- a/afs/afs/src/main/java/one/microstream/afs/types/AccessManager.java
+++ b/afs/afs/src/main/java/one/microstream/afs/types/AccessManager.java
@@ -905,7 +905,7 @@ public interface AccessManager
 			if(entry.removeShared(file) && entry.exclusive == null)
 			{
 				// if there is no more need for the entry itself, remove it.
-				this.fileUsers.removeFor(file);
+				this.fileUsers.removeFor(file.actual());
 			}
 		}
 		
@@ -936,6 +936,7 @@ public interface AccessManager
 		protected void removeExclusive(final AWritableFile file, final FileEntry entry)
 		{
 			entry.exclusive = null;
+			this.fileUsers.removeFor(file.actual());
 			file.retire();
 		}
 				

--- a/storage/storage/src/main/java/one/microstream/storage/types/StorageBackupHandler.java
+++ b/storage/storage/src/main/java/one/microstream/storage/types/StorageBackupHandler.java
@@ -33,7 +33,6 @@ import one.microstream.storage.exceptions.StorageExceptionBackupEmptyStorageBack
 import one.microstream.storage.exceptions.StorageExceptionBackupEmptyStorageForNonEmptyBackup;
 import one.microstream.storage.exceptions.StorageExceptionBackupInconsistentFileLength;
 import one.microstream.storage.types.StorageBackupHandler.Default.ChannelInventory;
-import one.microstream.storage.types.StorageDataFileValidator.Creator;
 
 public interface StorageBackupHandler extends Runnable, StorageActivePart
 {
@@ -571,7 +570,7 @@ public interface StorageBackupHandler extends Runnable, StorageActivePart
 		final void closeAllDataFiles()
 		{
 			final DisruptionCollectorExecuting<StorageClosableFile> closer = DisruptionCollectorExecuting.New(file ->
-				StorageClosableFile.close(file, null)
+				file.close()
 			);
 			
 			for(final ChannelInventory channel : this.channelInventories)

--- a/storage/storage/src/main/java/one/microstream/storage/types/StorageChannelTaskShutdown.java
+++ b/storage/storage/src/main/java/one/microstream/storage/types/StorageChannelTaskShutdown.java
@@ -75,7 +75,7 @@ public interface StorageChannelTaskShutdown extends StorageChannelTask
 			 */
 
 			// may not be done before to give every channel a safe way to notice the processing progress
-			this.operationController.deactivate();
+			// this.operationController.deactivate();
 
 			// can / may never throw an exception
 			channel.reset();

--- a/storage/storage/src/main/java/one/microstream/storage/types/StorageChannelTaskShutdown.java
+++ b/storage/storage/src/main/java/one/microstream/storage/types/StorageChannelTaskShutdown.java
@@ -74,9 +74,6 @@ public interface StorageChannelTaskShutdown extends StorageChannelTask
 			 * so that the issuing shutdown method waits for the shutdown to actually complete.
 			 */
 
-			// may not be done before to give every channel a safe way to notice the processing progress
-			// this.operationController.deactivate();
-
 			// can / may never throw an exception
 			channel.reset();
 		}

--- a/storage/storage/src/main/java/one/microstream/storage/types/StorageLiveFile.java
+++ b/storage/storage/src/main/java/one/microstream/storage/types/StorageLiveFile.java
@@ -204,7 +204,7 @@ public interface StorageLiveFile<S extends StorageLiveFile<S>> extends StorageCl
 				}
 			}
 			
-			throw new StorageException(StorageFileUser.class.getSimpleName() + " not found " + fileUser);
+			return false;
 		}
 				
 		@Override

--- a/storage/storage/src/main/java/one/microstream/storage/types/StorageSystem.java
+++ b/storage/storage/src/main/java/one/microstream/storage/types/StorageSystem.java
@@ -23,6 +23,9 @@ package one.microstream.storage.types;
 import static one.microstream.X.mayNull;
 import static one.microstream.X.notNull;
 
+import java.util.concurrent.atomic.AtomicBoolean;
+import java.util.concurrent.atomic.AtomicLong;
+
 import one.microstream.afs.types.AFileSystem;
 import one.microstream.meta.XDebug;
 import one.microstream.persistence.types.Persistence;
@@ -30,9 +33,6 @@ import one.microstream.persistence.types.Unpersistable;
 import one.microstream.storage.exceptions.StorageExceptionInitialization;
 import one.microstream.storage.exceptions.StorageExceptionNotAcceptingTasks;
 import one.microstream.storage.exceptions.StorageExceptionNotRunning;
-
-import java.util.concurrent.atomic.AtomicBoolean;
-import java.util.concurrent.atomic.AtomicLong;
 
 // (21.03.2016 TM)TODO: what is the difference between ~Manager and ~Controller here? Merge into Controller or comment.
 public interface StorageSystem extends StorageController
@@ -109,11 +109,11 @@ public interface StorageSystem extends StorageController
 		private final boolean                              switchByteOrder               ;
 
 		// state flags //
-		private AtomicBoolean    isStartingUp       = new AtomicBoolean();
-		private AtomicBoolean    isShuttingDown     = new AtomicBoolean();
+		private final AtomicBoolean    isStartingUp       = new AtomicBoolean();
+		private final AtomicBoolean    isShuttingDown     = new AtomicBoolean();
 		private final    Object  stateLock          = new Object();
-		private AtomicLong       initializationTime = new AtomicLong();
-		private AtomicLong       operationModeTime  = new AtomicLong();
+		private final AtomicLong       initializationTime = new AtomicLong();
+		private final AtomicLong       operationModeTime  = new AtomicLong();
 
 		// running state members //
 		private volatile StorageTaskBroker    taskbroker    ;
@@ -518,10 +518,9 @@ public interface StorageSystem extends StorageController
 				return;
 			}
 			
-			//the backup handler must shutdown first as it depends on the operationController
-			this.shutdownBackup();
 			
-//			DEBUGStorage.println("shutting down ...");
+			
+			
 			final StorageChannelTaskShutdown task = this.taskbroker.issueChannelShutdown(this.operationController);
 			
 			synchronized(task)
@@ -530,6 +529,10 @@ public interface StorageSystem extends StorageController
 				task.waitOnCompletion();
 			}
 			this.taskbroker = null;
+			
+			this.shutdownBackup();
+			
+			this.operationController.deactivate();
 
 			/* (07.03.2019 TM)FIXME: Shutdown must wait for ongoing activities.
 			 * Such as a StorageBackupHandler thread with a non-empty item queue.

